recipes/rangesnext/all/conanfile.py

```diff
@@ -0,0 +1,51 @@
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+import glob
+
+
+class RangesnextConan(ConanFile):
+    name = "rangesnext"
+    description = "ranges features for C++23 ported to C++20"
+    topics = ("conan", "rangesnext", "ranges", "backport", "backport-cpp")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/cor3ntin/rangesnext"
+    license = "BSL-1.0"
+    exports_sources = "patches/**"
+    settings = "compiler"
+
+    _compilers_minimum_version = {
+        "gcc": "10",
+        "Visual Studio": "19",
+        "clang": "13"
+    }
+    _source_subfolder = "source_subfolder"
+
+    def configure(self):
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, "20")
+
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if not minimum_version or tools.Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration("rangesnext requires C++20, which your compiler does not fully support.")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = glob.glob(self.name + "-*/")[0]
+        os.rename(extracted_dir, self._source_subfolder)
```

Comment:
```suggestion
        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
```

Since Conan 1.33.0 we can simplify it with `strip_root`: https://docs.conan.io/en/latest/changelog.html#jan-2021